src/rgw/driver/rados/rgw_user.cc

```diff
@@ -137,8 +137,13 @@ static void dump_user_info(Formatter *f, RGWUserInfo &info,
   encode_json("max_buckets", (int)info.max_buckets, f);
 
   dump_subusers_info(f, info);
-  dump_access_keys_info(f, info);
-  dump_swift_keys_info(f, info);
+
+  // keys cannot be viewed unless keys cap is set to read
+  int keys_perm = info.caps.check_cap("keys", RGW_CAP_READ);
+  if(keys_perm != -EPERM) {
```

Comment:
this needs to check the caps of the same `s->user` that sent the request; that's not necessarily the same user we're dumping the info for - an admin user could call `RGWOp_User_Info` with any value for `"uid"`

i think each `Op::execute()` function should do that `check_cap("keys", RGW_CAP_READ)` against `s->user`, and pass a `bool dump_keys` through functions like `RGWUserAdminOp_User::info()` to control whether we dump keys here